DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/README.md

```diff
@@ -0,0 +1,261 @@
+# Read-Only Cache 
+This FPGA tutorial demonstrates how to use the read-only cache feature to boost
+the throughput of an FPGA DPC++ design that uses a read-only look-up table
+defined in host code.
+
+***Documentation***:  The [DPC++ FPGA Code Samples
+Guide](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html)
+helps you to navigate the samples and build your knowledge of DPC++ for FPGA.
+<br>
+The [oneAPI DPC++ FPGA Optimization
+Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide)
+is the reference manual for targeting FPGAs through DPC++. <br>
+The [oneAPI Programming
+Guide](https://software.intel.com/en-us/oneapi-programming-guide) is a general
+resource for target-independent DPC++ programming.
+
+| Optimized for                     | Description
+---                                 |---
+| OS                                | Linux* Ubuntu* 18.04/20.04, RHEL*/CentOS* 8, SUSE* 15; Windows* 10
+| Hardware                          | Intel® Programmable Acceleration Card (PAC) with Intel Arria® 10 GX FPGA <br> Intel® FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix® 10 SX) <br> Intel® FPGA 3rd party / custom platforms with oneAPI support <br> *__Note__: Intel® FPGA PAC hardware is only compatible with Ubuntu 18.04* 
+| Software                          | Intel® oneAPI DPC++ Compiler <br> Intel® FPGA Add-On for oneAPI Base Toolkit 
+| What you will learn               | How and when to use the read-only cache feature
+| Time to complete                  | 30 minutes
+
+
+
+## Purpose
+
+This FPGA tutorial demonstrates an example of using the read-only cache to
+boost the throughput of an FPGA DPC++ design. Specifically, the read-only cache
+is most appropriate for table lookups that are constant throughout the
+execution of a kernel and is optimized for high cache hit performance.
+
+To enable the read-only cache, the `-Xsread-only-cache-size<N>` flag should be
+passed to the `dpcpp` command. Each kernel will get its own *private* version
+of the cache that serves all reads in the kernel from read-only no-alias
+accessors. Read-only no-alias accessors are accessors that have both the
+`read_only` and the `no_alias` properties:
+```c++
+accessor sqrt_lut(sqrt_lut_buf, h, read_only, accessor_property_list{no_alias});
+```
+The `read_only` property is required because the cache is *read_only*. The
+`no_alias` property is required to guarantee that the buffer will not be
+written to from the kernel through another accessor or through a USM pointer.
+Note that the same no-alias behavior can be achieved using the kernel attribute
+`[[intel::kernel_args_restrict]]`, when applicable.
```

Comment:
I don't understand this 'Note'.  Does that mean I don't need the no_alias property if I have the kernel_args_restrict property?